File: solutions/problem-20/FactorialDigitSum.java
/**
 *
 * @author namhsuyA
 */
public class FactorialDigitSum {
    private final int digit;

    public FactorialDigitSum(int digit) {
        this.digit = digit;
    }
    
    /**
     * 
     * @return factorial sum
     */
    public int getFactorialDigitSum(){
        //variable to store factorial
        int var = 1;
        String digits;
        for (int i = 1; i <= digit; i++) {
            var*=i;
        }
        digits = String.valueOf(var);
        
        //re initializing variable to store factorial sum
        var=0;
        for (String string : digits.split("")) {
            var+=Integer.parseInt(string);
        }
        return var;
    }   
}

File: solutions/problem-20/factorial_digit_sum.py
def factorial(fact_num = 100):
    f = 1

    if fact_num < 0:
       return -1
    elif fact_num == 0:
       return 1
    else:
       for i in range(1, fact_num + 1):
           f = f*i
       return f


def sum_of_digits(n):
    sum = 0
    while (n > 0):
        sum += n%10
        n = n//10

    return sum

print(sum_of_digits(factorial(100)))